src/components/builds/BuildsTable.tsx

```diff
@@ -0,0 +1,245 @@
+import { useRecoilValue } from 'recoil';
+import { graphql } from 'babel-plugin-relay/macro';
+
+import { WithStyles } from '@mui/styles';
+import { Link } from '@mui/material';
+import { createTheme } from '@mui/material/styles';
+import Table from '@mui/material/Table';
+import Stack from '@mui/material/Stack';
+import Tooltip from '@mui/material/Tooltip';
+import TableRow from '@mui/material/TableRow';
+import TableHead from '@mui/material/TableHead';
+import TableBody from '@mui/material/TableBody';
+import TableCell from '@mui/material/TableCell';
+import withStyles from '@mui/styles/withStyles';
+import InfoIcon from '@mui/icons-material/Info';
+import CommitIcon from '@mui/icons-material/Commit';
+import Typography from '@mui/material/Typography';
+import createStyles from '@mui/styles/createStyles';
+import CallSplitIcon from '@mui/icons-material/CallSplit';
+
+import BuildStatusChipNew from '../chips/BuildStatusChipNew';
+import { muiThemeOptions } from '../../cirrusTheme';
+import { shorten } from '../../utils/text';
+import { absoluteLink } from '../../utils/link';
+import { formatDuration } from '../../utils/time';
+import { navigateBuildHelper } from '../../utils/navigateHelper';
+
+import { BuildsTable_build } from './__generated__/BuildsTable_build.graphql';
+import { BuildsTable_repository } from './__generated__/BuildsTable_repository.graphql';
+
+const styles = theme =>
+  createStyles({
+    table: {
+      tableLayout: 'auto',
+    },
+    row: {
+      cursor: 'pointer',
+      '&.Mui-selected': {
+        background: `${
+          theme.palette.mode === 'dark' ? theme.palette.grey['800'] : theme.palette.grey['100']
+        } !important`,
+      },
+    },
+    cell: {
+      fontSize: 16,
+      whiteSpace: 'nowrap',
+      textOverflow: 'ellipsis',
+      '& *': {
+        fontSize: '16px !important',
+      },
+    },
+    cellStatus: {
+      width: 160,
+      minWidth: 160,
+      maxWidth: 160,
+    },
+    cellStatusChip: {
+      '& *': {
+        fontSize: '14px !important',
+      },
+    },
+    cellCommit: {},
+    cellHash: {
+      width: 140,
+      minWidth: 140,
+      maxWidth: 140,
+    },
+    cellBranch: {
+      width: 210,
+      minWidth: 210,
+      maxWidth: 210,
+    },
+    cellDuration: {
+      width: 100,
+      minWidth: 100,
+      maxWidth: 100,
+      textAlign: 'right',
+    },
+    infoIcon: {
+      color: theme.palette.action.active,
+    },
+  });
+
+const styled = withStyles(styles);
+
+interface Props extends WithStyles<typeof styles> {
+  builds: Array<BuildsTable_build>;
+  repository: BuildsTable_repository;
+  selectedBuildId?: string;
+  setSelectedBuildId: Function;
+}
+
+const BuildsTable = styled(({ classes, builds = [], repository, selectedBuildId, setSelectedBuildId }: Props) => {
+  const themeOptions = useRecoilValue(muiThemeOptions);
+  const muiTheme = useMemo(() => createTheme(themeOptions), [themeOptions]);
+
+  return (
+    <ThemeProvider theme={muiTheme}>
```

Comment:
This component should already be under a theme provider. I guess it might work without it here.

Reply:
The idea is to use default MUI theme as much as possible, applying adjustments when needed, and get rid of the current `cirrusTheme.ts` with its style overrides.

But for now, as we are making redesign step by step, I'm planning to keep two themes at the same time: `currisTheme` for the old design and `muiTheme` for the new design.

Whenever we want to apply new design theme, the component should be wrapped into `<ThemeProvider theme={muiTheme}>`. When the new design will be implemented for the whole app, `cirrusTheme` will be dropped with the all temporary `ThemeProvider` wrappers (like this one).

Reply:
I can't reuse `cirrusTheme` as it affects default MUI styles (e.g. table paddings). If I keep these overridden styles, then the new design is broken, but removing them brakes the old one.

Reply:
Nice idea! 🚀